docs/source/tutorial/checkpoints.rst

```diff
@@ -0,0 +1,147 @@
+Using Checkpoints
+=================
+Training may take days to weeks in extreme cases when using models with many parameters or big datasets. This introduces
+a large array of possible errors, e.g. session timeouts, server restarts etc., which would lead to a complete loss of
+all progress made so far. To avoid this the :class:`pykeen.training.TrainingLoop` supports built-in check-points that
+allow a straight-forward saving of the current training loop state and resumption of a saved
+state from saved checkpoints.
+
+How to do it
+------------
+To show how checkpoints are used with PyKEEN let's look at a simple example of how a model is setup.
+For fixing possible errors and safety fallbacks please also look at :ref:`word_of_caution`.
+
+.. code-block:: python
```

Comment:
wouldn't these tutorials be more useful for users if they started by being centered on the `pipeline` and then at the end gave some insight into the underlying implementation?

Reply:
The reason it is kept right now is due to the fact that the checkpoint functionality is only a *true* training loop functionality, because even though the pipeline supports using training loop checkpoints, it is not a *true* pipeline checkpoint.

Reply:
That's true, but I don't think the beginning of a tutorial section of the documentation benefits from being pedagogical. The technical part can be in the reference, or at the end of the tutorial to help users who understand how to use the simple parts and want to understand how it works. I think one place this worked really well was the [First Steps](https://pykeen.readthedocs.io/en/latest/tutorial/first_steps.html) tutorial, which ended with the [Beyond the Pipeline](https://pykeen.readthedocs.io/en/latest/tutorial/first_steps.html#beyond-the-pipeline) section.

Could you elaborate on the difference you mean between a training loop checkpoint vs a pipeline checkpoint?